internal/service/dynamodb/global_secondary_index_test.go

```diff
@@ -0,0 +1,2248 @@
+					acctest.ImportCheckResourceAttr(acctest.CtTagsPercent, "0"),
+					acctest.ImportCheckResourceAttr(acctest.CtTagsAllPercent, "0"),
+					acctest.ImportCheckResourceAttr("read_capacity", "0"),
+					acctest.ImportCheckResourceAttr("write_capacity", "0"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.#", "1"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.0.%", "2"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.0.max_read_request_units", "2"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.0.max_write_request_units", "2"),
+				),
+			},
+			{
+				ResourceName:      resourceNameGSI,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"read_capacity",
+					"write_capacity",
+				},
+				ImportStateCheck: acctest.ComposeAggregateImportStateCheckFunc(
+					acctest.ImportCheckResourceAttr(names.AttrName, rNameGSI),
+					acctest.ImportCheckResourceAttr("table", rNameTable),
+					acctest.ImportCheckResourceAttr("hash_key", rNameTable),
+					acctest.ImportCheckResourceAttr("hash_key_type", "S"),
+					acctest.ImportCheckResourceAttr("range_key", rNameGSI),
+					acctest.ImportCheckResourceAttr("range_key_type", "S"),
+					acctest.ImportCheckResourceAttr("projection_type", "ALL"),
+					acctest.ImportCheckResourceAttr("read_capacity", "0"),  // inherited from table
+					acctest.ImportCheckResourceAttr("write_capacity", "0"), // inherited from table
+					acctest.ImportCheckResourceAttr("on_demand_throughput.#", "1"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.0.%", "2"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.0.max_read_request_units", "2"),
+					acctest.ImportCheckResourceAttr("on_demand_throughput.0.max_write_request_units", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckInitialGSIExists(n string, tbl *awstypes.TableDescription, gsi *awstypes.GlobalSecondaryIndexDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if tbl == nil {
+			return errors.New("Table config is empty")
+		}
+
+		for _, g := range tbl.GlobalSecondaryIndexes {
+			if rs.Primary.Attributes[names.AttrName] == aws.ToString(g.IndexName) {
+				*gsi = g
+
+				break
+			}
+		}
+
+		return nil
+	}
+}
+
+func testAccGlobalSecondaryIndexConfig_basic(tableName, indexName string) string {
+	return fmt.Sprintf(`
+resource "aws_dynamodb_table" "test" {
+  name           = %[1]q
+  hash_key       = %[1]q
+  read_capacity  = 1
+  write_capacity = 1
+
+  attribute {
+    name = %[1]q
+    type = "S"
+  }
+}
+
+resource "aws_dynamodb_global_secondary_index" "test" {
+  table           = aws_dynamodb_table.test.name
+  name            = %[2]q
+  hash_key        = %[1]q
+  range_key       = %[2]q
+  read_capacity   = 1
+  write_capacity  = 1
+  range_key_type  = "S"
+  projection_type = "ALL"
+}
```

Comment:
Since the `aws_dynamodb_global_secondary_index` is the resource type being tested, move it to the top of the configuration